istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,564 @@
+			"If no local file is specified, the CA certificate will be read from default service account. "+
+			"The CA certificate will be verified against the webhook certificate specified in --valiation-secret.")
+	flags.StringVar(&opts.webhookConfigPath, "config-path", "",
+		"Specifies the file path of the webhook configuration.")
+	flags.StringVar(&opts.validatingWebhookConfigName, "config-name", "istio-galley",
+		"The name of the ValidatingWebhookConfiguration to manage.")
+	flags.StringVar(&opts.validatingWebhookServiceName, "service", "istio-galley",
+		"The service name of the validating webhook to manage.")
+	flags.StringVar(&opts.validationSecretName, "validation-secret", "",
+		"The name of the secret of the validation webhook. istioctl will verify that the webhook certificate "+
+			"is issued by the CA certificate.")
+	flags.DurationVar(&opts.maxTimeForCheckingWebhookServer, "timeout", 60*time.Second,
+		"	Max time for checking the validating webhook server. If the validating webhook server is not ready"+
+			"in the given time, exit. Otherwise, apply the webhook configuration.")
+	flags.DurationVar(&opts.maxTimeForReadingWebhookCert, "read-cert-timeout", 60*time.Second,
+		"	Max time for waiting the webhook certificate to be readable.")
+
+	return cmd
+}
+
+func newDisableCmd() *cobra.Command {
+	opts := &disableCliOptions{}
+	cmd := &cobra.Command{
+		Use:   "disable",
+		Short: "Disable webhook configurations",
+		Example: `
+# Disable the webhook configuration of Galley
+istioctl experimental post-install webhook disable --validation --validation-config istio-galley
+# Disable the webhook configuration of Galley and Sidecar Injector
+istioctl experimental post-install webhook disable --validation --validation-config istio-galley --injection --injection-config istio-sidecarinjector
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				return fmt.Errorf("err when creating k8s client interface: %v", err)
+			}
+			err = disableWebhookConfig(client, opts)
+			if err != nil {
+				return fmt.Errorf("error when disabling webhook configurations: %v", err)
+			}
+			fmt.Println("webhook configurations have been disabled")
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.disableValidationWebhook, "validation", false, "Specifies whether disabling"+
+		"the validating webhook.")
+	flags.StringVar(&opts.validatingWebhookConfigName, "validation-config", "istio-galley",
+		"The name of the ValidatingWebhookConfiguration to disable.")
+	flags.BoolVar(&opts.disableInjectionWebhook, "injection", false, "Specifies whether disabling"+
+		"the validating webhook.")
+	flags.StringVar(&opts.mutatingWebhookConfigName, "injection-config", "istio-sidecarinjector",
+		"The name of the MutatingWebhookConfiguration to disable.")
```

Comment:
Just `The mutating webhook configuration to disable`.

Reply:
Done.